README.md

```diff
@@ -0,0 +1,59 @@
+# CTile
+
+This is a caching proxy for the get-entries endpoint of a CT log, which uses S3
+as its backing store. It uses the concept of "tiles" of entries, where each tile
+is a fixed size N (e.g. 256 log entries) and the sequence of tiles starts at 0.
+Regardless of what `start` and `end` parameters CTile receives for a request, it
+will transform those into a tile-sized request to its backend, by rounding down
+`start` to the nearest multiple of N and requesting exactly N items from the
+backend. If the request is successful, CTile checks that the response contains
+exactly N items, re-encodes as gzipped CBOR, and stores the result in S3. It then
+returns modified JSON to the user, removing items from the head and tail to ensure
+that the first entry actually corresponds to the first entry requested by the user
+and that the response includes at most as many entries as requested.
+
+When looking up entries in the cache, CTile also rounds `start` down to the
+nearest multiple of N, and requests a single tile from the S3 backend. The CT
+protocol allows the server to return fewer results than requested, so CTile does
+not attempt to request multiple tiles to fulfil a large request. If a request's
+`start` parameter is one less than the end of a tile, CTile will respond with a
+single entry. This is similar to how Trillian's [align_getentries
+flag](https://github.com/google/certificate-transparency-go/blob/6e118585d9d9757b739353829becec378f47e10b/trillian/ctfe/handlers.go#L50)
+works, and is in fact compatible with that flag, so long as CTile's tile size is
+equal to Trillian's max_get_entries flag.
```

Comment:
```suggestion
single entry. This is similar to how Trillian's [align_getentries
flag](https://github.com/google/certificate-transparency-go/blob/6e118585d9d9757b739353829becec378f47e10b/trillian/ctfe/handlers.go#L50)
works, and is in fact compatible with that flag, so long as CTile's tile size is
less than or equal to Trillian's max_get_entries flag.
```

The one simplification that CTile makes to the logic is that you can only receive entries from the tile containing the `start` of the request, whereas in Trillian, you can get back entries from two different tiles if your `start` and `end` values cross a boundary (but only if `end-start+1 <= max_get_entries`). However, since this logic is at least as strict as Trillian's, I believe it's correct to say that it's compatible for any tile size *less than or equal to* Trillian's max_get_entries.

Reply:
(That's not to say it's a particularly good idea to go setting these differently, because that could get really confusing really quickly)

Reply:
Oh interesting! Yeah, looking at the code and comments (https://github.com/google/certificate-transparency-go/blob/f6571bdcb6b6ad9972248e724ff447edf664ce45/trillian/ctfe/handlers.go#L973-L987) it appears that align_get_entries only applies to "maximally sized" get-entries. If you request smaller batches, you don't get coerced to tile boundaries.

So you're right, less than or equal is correct here. Thanks for spotting!